frontend/public/components/utils/dropdown.jsx

```diff
@@ -285,7 +285,18 @@ class Dropdown_ extends DropdownMixin {
   applyTextFilter_(autocompleteText, items) {
     const { autocompleteFilter } = this.props;
     if (autocompleteFilter && !_.isEmpty(autocompleteText)) {
-      items = _.pickBy(items, (item, key) => autocompleteFilter(autocompleteText, item, key));
+      const newItems = [];
+      let totalItems = _.map(items, (item) => item.props);
+      totalItems = autocompleteFilter(autocompleteText, totalItems);
+      _.map(totalItems, (filterItem) => {
+        _.some(items, (item) => {
+          if (item.props.name === filterItem.target) {
```

Comment:
Callers need to be able to filter on properties other than name.

Reply:
By what I understood from the code where dropdown is used that the object that is passed to this Component has the same structure everywhere and filters based on name only. I couldn't find any other case, so kept it as it is.